### PR TITLE
Closes #1394: Updating views re-order notice blocks to use render arrays and Url::fromRoute.

### LINKED ISSUE #1394: Carousel notification links broken for sites not installed at domain basepath
## Problem/Motivation
My department hosts sites in a multisite configuration where some sites are installed at subdirectory paths instead of the domain basepath.

### Describe the bug
The Carousel notification message contains links that assume a site is installed at the local basepath / such as example.arizona.edu, creating a link at /admin/content?title=&type=az_carousel_item.

But sites can exist in a subdirectory path such as example.arizona.edu/mysite, and the links in the notification don't account for that. They continue to point to /admin/content?title=&type=az_carousel_item instead of the correct link of /mysite/admin/content?title=&type=az_carousel_item

Update: The Reorder People block also has the same issue.

### To Reproduce
Install a new AZQS site in a subdirectory path and enable Quickstart Carousel Item module to get the AZ Carousel Notice block. Click a link in the notification block and it will not work as expected.

## Proposed resolution
Use [base_path()](https://api.drupal.org/api/drupal/core%21includes%21common.inc/function/base_path/9.0.x) instead of / for the links.

Also possible to include here: change the return to use [t()](https://api.drupal.org/api/drupal/core%21includes%21bootstrap.inc/function/t/9.0.x) to support translations and use placeholders for links.
Guidance that might still be relevant: [Dynamic or static links and HTML in translatable strings
](https://www.drupal.org/docs/7/api/localization-api/dynamic-or-static-links-and-html-in-translatable-strings)

### Expected behavior
When on a site installed at a subdirectory path, clicking the links in the Carousel notification block should work.

### PR DESCRIPTION
**Pull request by trackleft**
## Related Issue
Closes #1394 

## How Has This Been Tested?
Locally

## Types of changes
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

**Drupal core**
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

**Drupal contrib projects**
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist:
- [x] My code follows the code style of this project.
- [] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by trackleft** [comment]
@joshuasosa added an alternative approach that we should consider.
https://www.drupal.org/project/draggableviews/issues/2843858#:~:text=commented%208%20months-,ago,-I%20know%20this

File: modules/custom/az_person/src/Plugin/Block/AZPersonReorderNotice.php [hunk 1]
```diff
@@ -23,9 +24,41 @@ class AZPersonReorderNotice extends BlockBase {
    */
   public function build() {
 
-    return [
-      '#markup' => '<div class="callout callout-leaf"><p><strong>Need to reorder people?</strong></p><p>Use the button below to reorder the people on this page.</p><p><a class="btn btn-success" href="/admin/az-person/reorder-people">Reorder People</a></p></div>',
+    $callout_markup_header = [
+      '#markup' => '<p><strong>' . $this->t('Need to reorder people?') . '</strong></p>',
     ];
+    $callout_markup_description = [
+      '#markup' => '<p>' . t('Use the button below to reorder the people on this page.') . '</p>',
```
Comment (trackleft) [inline modules/custom/az_person/src/Plugin/Block/AZPersonReorderNotice.php:31]:
```suggestion
      '#markup' => '<p>' . $this->t('Use the button below to reorder the people on this page.') . '</p>',
```

Comment (trackleft) [inline modules/custom/az_person/src/Plugin/Block/AZPersonReorderNotice.php:31]:
```suggestion
      '#markup' => '<p>' . $this->t('Use the button below to reorder the people on this page.') . '</p>',
```

**Review by akslay — DISMISSED** [review]

File: modules/custom/az_person/src/Plugin/Block/AZPersonReorderNotice.php (hunk 1 above)
Comment (joeparsons) [reply modules/custom/az_person/src/Plugin/Block/AZPersonReorderNotice.php:31]: FWIW, I'd be in favor of replacing the blocks with contextual links, if possible.

**Review by joeparsons — APPROVED** [review]
I'd be OK with merging this PR now and then look into replacing these blocks with contextual links as a follow-up.

**Review by joeparsons — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]
Confirmed in Probo that this at least does not cause a regression and the Reorder People link still takes you to the correct page.

**Comment by joeparsons** [comment]
Follow-up issue for replacing blocks with contextual links: #1449

**Comment by joshuasosa** [comment]
Apologies for the late notice (forgot to update notification settings). Thanks for helping to resolve this issue! I agree, moving these links to contextual links is a good idea too!

**Pull request merged**